app/screens/home/search/results/results.tsx

```diff
@@ -65,207 +60,57 @@ const SearchResults = ({
     loading,
     posts,
     publicLinkEnabled,
+    scrollPaddingTop,
     searchValue,
     selectedTab,
 }: Props) => {
+    const dimensions = useWindowDimensions();
     const theme = useTheme();
-    const isTablet = useIsTablet();
-    const insets = useSafeAreaInsets();
-    const [lastViewedIndex, setLastViewedIndex] = useState<number | undefined>(undefined);
-
-    const orderedPosts = useMemo(() => selectOrderedPosts(posts, 0, false, '', '', false, isTimezoneEnabled, currentTimezone, false).reverse(), [posts]);
-    const {images: imageAttachments, nonImages: nonImageAttachments} = useImageAttachments(fileInfos, publicLinkEnabled);
-    const channelNames = useMemo(() => fileChannels.reduce<{[id: string]: string | undefined}>((acc, v) => {
-        acc[v.id] = v.displayName;
-        return acc;
-    }, {}), [fileChannels]);
-
-    const containerStyle = useMemo(() => {
-        let padding = 0;
-        if (selectedTab === TabTypes.MESSAGES) {
-            padding = posts.length ? 4 : 8;
-        } else {
-            padding = fileInfos.length ? 8 : 0;
-        }
-        return {top: padding};
-    }, [selectedTab, posts, fileInfos]);
-
-    const filesForGallery = useMemo(() => imageAttachments.concat(nonImageAttachments),
-        [imageAttachments, nonImageAttachments]);
-
-    const orderedFilesForGallery = useMemo(() => (
-        filesForGallery.sort((a: FileInfo, b: FileInfo) => {
-            return (b.create_at || 0) - (a.create_at || 0);
-        })
-    ), [filesForGallery]);
-
-    const filesForGalleryIndexes = useMemo(() => orderedFilesForGallery.reduce<{[id: string]: number | undefined}>((acc, v, idx) => {
-        if (v.id) {
-            acc[v.id] = idx;
-        }
-        return acc;
-    }, {}), [orderedFilesForGallery]);
-
-    const handlePreviewPress = useCallback(preventDoubleTap((idx: number) => {
-        const items = orderedFilesForGallery.map((f) => fileToGalleryItem(f, f.user_id));
-        openGalleryAtIndex(galleryIdentifier, idx, items);
-    }), [orderedFilesForGallery]);
-
-    const snapPoints = useMemo(() => {
-        let numberOptions = 1;
-        if (canDownloadFiles) {
-            numberOptions += 1;
-        }
-        if (publicLinkEnabled) {
-            numberOptions += 1;
-        }
-        return [bottomSheetSnapPoint(numberOptions, ITEM_HEIGHT, insets.bottom) + HEADER_HEIGHT, 10];
-    }, [canDownloadFiles, publicLinkEnabled]);
-
-    const handleOptionsPress = useCallback((item: number) => {
-        setLastViewedIndex(item);
-        const renderContent = () => {
-            return (
-                <FileOptions
-                    fileInfo={orderedFilesForGallery[item]}
-                />
-            );
+    const styles = getStyles(dimensions);
```

Comment:
or use here `useMemo`

Reply:
I like this to allow keeping `getStyles` outside the component.